pkg/pillar/cmd/volumemgr/updatestatus.go

```diff
@@ -259,6 +260,15 @@ func doUpdateVol(ctx *volumemgrContext, status *types.VolumeStatus) (bool, bool)
 			changed = true
 			// Work is done
 			DeleteWorkCreate(ctx, status)
+			if status.MaxVolSize == 0 {
+				var err error
+				log.Infof("doUpdateVol: MaxVolSize is 0 for %s. Filling it up.",
+					status.FileLocation)
+				_, status.MaxVolSize, err = utils.GetVolumeSize(status.FileLocation)
+				if err != nil {
+					log.Error(err)
+				}
+			}
```

Comment:
Don't we also need this logic when we find a volume after reboot and set the state to VOLUME_CREATED?

Reply:
Yes, we set the state to VOLUME_CREATED after device reboot.